pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,345 @@
+					ConntrackCommitTable,
+				},
+				binding.OutputStage: {
+					L2ForwardingOutTable,
+				},
+			},
+		}
+		if f.enableMulticast {
+			template.addTable(binding.ValidationStage, IPClassifierTable)
+		}
+		for _, ipProtocol := range f.ipProtocols {
+			if ipProtocol == binding.ProtocolIPv6 {
+				template.addTable(binding.ValidationStage, IPv6Table)
+				break
+			}
+		}
+	case pipelineARP:
+		template = &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.ValidationStage: {
+					ARPSpoofGuardTable,
+				},
+				binding.OutputStage: {
+					ARPResponderTable,
+				},
+			},
+		}
+	}
+
+	return template
+}
+
+func (f *featureNetworkPolicy) getTemplate(p pipeline) *featureTemplate {
+	if p != pipelineIP {
+		return nil
+	}
+	template := &featureTemplate{
+		stageTables: map[binding.StageID][]*FeatureTable{
+			binding.EgressSecurityStage: {
+				EgressRuleTable,
+				EgressDefaultTable,
+				EgressMetricTable,
+			},
+			binding.IngressSecurityStage: {
+				IngressSecurityClassifierTable,
+				IngressRuleTable,
+				IngressDefaultTable,
+				IngressMetricTable,
+			},
+		},
+	}
+	if f.enableAntreaPolicy {
+		template.addTable(binding.EgressSecurityStage, AntreaPolicyEgressRuleTable)
+		template.addTable(binding.IngressSecurityStage, AntreaPolicyIngressRuleTable)
+	}
+	return template
+}
+
+func (f *featureService) getTemplate(p pipeline) *featureTemplate {
+	if p != pipelineIP {
+		return nil
+	}
+	if !f.enableProxy {
+		return &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.PreRoutingStage: {
+					DNATTable,
+				},
+			},
+		}
+	}
+	template := &featureTemplate{
+		stageTables: map[binding.StageID][]*FeatureTable{
+			binding.ConntrackStateStage: {
+				SNATConntrackTable,
+			},
+			binding.PreRoutingStage: {
+				PreRoutingClassifierTable,
+				SessionAffinityTable,
+				ServiceLBTable,
+				EndpointDNATTable,
+			},
+			binding.RoutingStage: {
+				L3ForwardingTable,
+				ServiceHairpinMarkTable,
+			},
+			binding.PostRoutingStage: {
+				SNATConntrackCommitTable,
+			},
+			binding.ConntrackStage: {
+				ConntrackCommitTable,
+			},
+			binding.OutputStage: {
+				L2ForwardingOutTable,
+			},
+		},
+	}
+	if f.proxyAll {
+		template.addTable(binding.PreRoutingStage, NodePortMarkTable)
+	}
+	return template
+}
+
+func (f *featureEgress) getTemplate(p pipeline) *featureTemplate {
+	if p != pipelineIP {
+		return nil
+	}
+	return &featureTemplate{
+		stageTables: map[binding.StageID][]*FeatureTable{
+			binding.RoutingStage: {
+				L3ForwardingTable,
+			},
+			binding.PostRoutingStage: {
+				SNATTable,
+			},
+		},
+	}
+}
+
+func (f *featureMulticast) getTemplate(p pipeline) *featureTemplate {
+	if p != pipelineMulticast {
+		return nil
+	}
+	return &featureTemplate{
+		stageTables: map[binding.StageID][]*FeatureTable{
+			binding.RoutingStage: {
+				MulticastTable,
+			},
+		},
+	}
+}
+
+// tracedFeature is the interface to provide flows for Traceflow.
```

Comment:
tracedFeature is the interface intended to support feature Traceflow. Any other feature expected to trace the packet status with its flow entries needs to implement this interface.

Reply:
flowsToTrace returned the flows to be installed when a packet tracing request is created.

Reply:
Done.